src/components/Offers/Edit/EditOfferForm.js

```diff
@@ -0,0 +1,169 @@
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import { parseRequestErrors } from "../Form/OfferUtils";
+import OfferForm from "../Form/form-components/OfferForm";
+import { editOffer } from "../../../services/offerService";
+import { Redirect, useLocation, useParams } from "react-router-dom";
+import useOffer from "../../../hooks/useOffer";
+import useOfferForm from "../../../hooks/useOfferForm";
+import { INITIAL_JOB_DURATION } from "../../../reducers/searchOffersReducer";
+import useSession from "../../../hooks/useSession";
+import EditOfferSchema from "./EditOfferSchema";
+
+export const EditOfferControllerContext = React.createContext();
+
+function parseDescription(rawDescription) {
+    const temp = document.createElement("span");
+    temp.innerHTML = rawDescription;
+    const description = temp.textContent || temp.innerText;
+    temp.remove();
+    return description;
+}
+
+const parseOfferForm = ({
+    jobMinDuration,
+    jobMaxDuration,
+    requirements,
+    contacts,
+    jobStartDate,
+    isPaid,
+    vacancies,
+    description,
+    ...offer
+}) => ({
+    jobDuration: [
+        jobMinDuration || INITIAL_JOB_DURATION,
+        jobMaxDuration || INITIAL_JOB_DURATION + 1,
+    ],
+    isPaid: isPaid || "none",
+    requirements: requirements.map((value) => ({ value })),
+    contacts: contacts.map((value) => ({ value })),
+    jobStartDate: jobStartDate || null,
+    vacancies: vacancies || "",
+    description,
+    descriptionText: parseDescription(description),
+    ...offer,
+});
+
+export const EditOfferController = () => {
+    const { id } = useParams();
+    const { offer, error: errorOffer, loading: loadingOffer } = useOffer(id);
+    const { data: user, isValidating } = useSession();
+
+    // This portion of code is used to remove race conditions between useState of canEdit and useEffect
+    // If the value of useState is false by default, this condition will be wrongly verified, resulting in unwanted redirects
+    const condition = useCallback(() => {
```

Comment:
`condition` is too broad, use something like `shouldRevalidateEditingPermissions`